### PR TITLE
473 fix dag

### PR DESCRIPTION
**Pull request by hiker**
👕 👕 I can't say no, right?

This implements @rupertford's short term solution: ignore the start, stop, and step value of loops, and add the schedules with a unique name to the graphs in all tests.

### CODE REVIEW

File: src/psyclone/psyGen.py [hunk 1]
```diff
@@ -1136,8 +1136,14 @@ def dag_gen(self, graph):
             graph.edge(remote_name, local_name, color="blue")
         # now call any children so they can add their information to
         # the graph
-        for child in self.children:
-            child.dag_gen(graph)
+        if isinstance(self, Loop):
+            # Ignore the children that represents the start, stop
+            # and step value of the loop:
+            for child in self.children[3:]:
```
Comment (arporter) [inline src/psyclone/psyGen.py]: Please use `loop_body` here.

File: src/psyclone/tests/psyGen_test.py [hunk 2]
```diff
@@ -2366,15 +2374,17 @@ def test_node_dag(tmpdir, have_graphviz):
     schedule.dag(file_name=my_file.strpath)
     result = my_file.read()
     print(result)
```
Comment (arporter) [inline src/psyclone/tests/psyGen_test.py]: Please remove all three of these print statements.

File: src/psyclone/psyGen.py (hunk 1 above)
Comment (hiker) [reply src/psyclone/psyGen.py]: Oops - how embarrassing.

File: src/psyclone/tests/psyGen_test.py (hunk 2 above)
Comment (hiker) [reply src/psyclone/tests/psyGen_test.py]: Done

**Comment by hiker** [comment]
That should be ready for review - the failing coverage seems to be caused by travis not executing the dag tests.

**Comment by arporter** [comment]
All requested changes have been made. All tests pass/xfail when I have the graphviz package installed. I was on the point of accepting when I remembered documentation :-( But then, I've realised that my PR #580 also requires that that same documentation be updated so I'm going to be kind and accept this as is :-)

**Pull request merged**